File: Learn-Datastructures/C#/Basic DataStructures/Basic DataStructures/Stack.cs
public class Stack
{
    private List<int> stack = new List<int>();

    public void Push(int value)
    {
        stack.Add(value);
    }

    public int Pop()
    {
        if (stack.Count == 0) throw new InvalidOperationException("Stack is empty");
        stack.RemoveAt(stack.Count - 1);
    }

    public int Peek()
    {
        if (stack.Count == 0) throw new InvalidOperationException("Stack is empty");
        return stack[stack.Count - 1];
    }
}

File: Data structures/C/Stack.c
#include <stdio.h>
#define MAX 5

int stack[MAX];
int top = -1;

void push(int value) {
    if (top == MAX - 1)
        printf("Stack overflow\n");
    else
        stack[++top] = value;
}

int pop() {
    if (top == -1)
        printf("Stack underflow\n");
    else
        return stack[top--];
}

void display() {
    if (top == -1)
        printf("Stack is empty\n");
    else {
        for (int i = top; i >= 0; i--)
            printf("%d ", stack[i]);
        printf("\n");
    }
}

int main() {
    push(10);
    push(20);
    push(30);
    display();
    pop();
    display();
    return 0;
}
